### ISSUE TITLE
[java] UnsynchronizedStaticFormatter false-positive in static constructor

### ISSUE THREAD

**Issue by boris-petrov** [pmd/pmd#1895]
PMD version:
6.16.0+
7.0.0

Thank you for fixing the [other issue](https://github.com/pmd/pmd/issues/1814) with `UnsynchronizedStaticFormatter`!

However, the new version has a bug:

```java
public class Foo {
	private static final NumberFormat NUMBER_FORMAT = NumberFormat.getInstance();

	static {
		NUMBER_FORMAT.setGroupingUsed(false);
	}

...
```

This raises a warning while it shouldn't as using the variable unsynchronized in the static constructor is totally fine.

**Comment by mrclmh** [comment]
Not reproducible with 7.18.0.
This has been fixed with https://github.com/pmd/pmd/issues/2368 and can be closed.
